crates/driver/src/infra/config/file/mod.rs

```diff
@@ -574,3 +579,77 @@ pub enum GasEstimatorType {
     Native,
     Web3,
 }
+
+/// Defines various strategies to prioritize orders.
+#[derive(Debug)]
+pub enum OrderPriorityStrategy {
+    /// Strategy to prioritize orders based on their class. Market orders are
+    /// preferred over limit orders, as the expectation is that they should
+    /// be immediately fulfillable. Liquidity orders come last, as they are
+    /// the most niche and rarely used.
+    OrderClass,
+    /// Strategy to prioritize orders based on external price.
+    /// This strategy uses the likelihood that an order will be fulfilled,
+    /// based on token prices. A larger value means that the order is more
+    /// likely to be fulfilled.
+    ExternalPrice,
+    /// Strategy to prioritize orders based on their creation timestamp. The
+    /// most recently created orders are given the highest priority.
+    CreationTimestamp {
+        /// When specified, orders created within this threshold will be
+        /// prioritized.
+        ///
+        /// Example: creation-timestamp:120s
+        max_order_age: Option<Duration>,
+    },
+    /// Strategy to prioritize orders based on whether the current solver
+    /// provided the winning quote for the order.
+    OwnQuotes,
+}
+
+impl<'de> Deserialize<'de> for OrderPriorityStrategy {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        let parts: Vec<&str> = s.split(':').collect();
```

Comment:
Did you test which format is now needed to configure these values now? Asking because you initially didn't want to use plain serde because there were some limitations. But now it's reverted but the test doesn't have this configured anywhere so it could be that serde is able to parse the missing value but requires some weird format when you actually want to configure it.

Reply:
Fixed and checked with a unit test.

Reply:
Bit unfortunate that we have to specify an empty object for a few enums but I'd say that's okay. Thanks for checking.